src/Microsoft.AspNet.Razor/TagHelpers/TagHelperAttributeDescriptorComparer.cs

```diff
@@ -31,7 +31,8 @@ protected TagHelperAttributeDescriptorComparer()
         /// <remarks>
         /// Determines equality based on <see cref="TagHelperAttributeDescriptor.IsIndexer"/>,
         /// <see cref="TagHelperAttributeDescriptor.Name"/>, <see cref="TagHelperAttributeDescriptor.PropertyName"/>,
-        /// and <see cref="TagHelperAttributeDescriptor.TypeName"/>. Ignores
+        /// <see cref="TagHelperAttributeDescriptor.TypeName"/>, and
+        /// <see cref="TagHelperAttributeDescriptor.UsageDescriptor"/>. Ignores
```

Comment:
My comment assumes we never compare attribute descriptors except within tag helper descriptors. Where is that invalidated? If it's incorrect, a lot more code needs to change.